### ISSUE TITLE
Upgrade the TLS version to 1.2 or 1.3

### ISSUE THREAD

**Issue by moksharthab** [Velocidex/velociraptor#2364]
Hi team,

We have Qualys running in our environment and they have notified of couple of vulnerabilities:

SSL/TLS Server supports TLSv1.0
SSL/TLS Server supports TLSv1.1

We have checked the config file options and we dont see any option to enable it.

Can you please help on how we can upgrade to TLSv1.2 ?

Thanks in advance.

**Comment by moksharthab** [comment]
Or preferably upgrade to TLSv1.3

**Comment by predictiple** [comment]
Try `curl -v https://<your.server.name>`?

Mine shows TLSv1.3 ("SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256")

**Comment by scudette** [comment]
This is the configuration for the frontend TLS endpoint

For frontend
https://github.com/Velocidex/velociraptor/blob/8d63fc11d94422bfa0445ead0011b16459e82e15/api/builder.go#L335-L340

For GUI
https://github.com/Velocidex/velociraptor/blob/8d63fc11d94422bfa0445ead0011b16459e82e15/api/builder.go#L648-L655

But for the autocert configured server we just use the autocert library to get tls configuration which looks like it does not set minimum TLS versions
https://github.com/Velocidex/velociraptor/blob/8d63fc11d94422bfa0445ead0011b16459e82e15/api/builder.go#L500

https://cs.opensource.google/go/x/crypto/+/refs/tags/v0.5.0:acme/autocert/autocert.go;l=227

So I think this is the issue - we need to force the autocert TLS parameters to comply with the minimum settings while retaining the cert maneger (which is really what is required to manage the lets encrypt certs).

**Comment by moksharthab** [comment]
Thank you Mike.

Can you please help on how to gorce autocert tls parameters ?

Thanks in advance.

Regards,
Mokshartha

**Comment by moksharthab** [comment]
Gentle reminder team!

**Comment by scudette** [comment]
This was fixed in #2367 and will be in the next release.

**Comment by moksharthab** [comment]
Had a look through the repo and see releases are pretty frequent (monthly?) but couldn't find anything on target release date or ETA for next release which would contain the released fix for #2367, where can we find that release schedule information?

**Comment by scudette** [comment]
Our releases are probably about every few months and follow the process described here https://docs.velociraptor.app/docs/overview/support/

We also build test binaries at every commit with our CI pipeline (https://github.com/Velocidex/velociraptor#getting-the-latest-version). If you want to test that it fixes the issue you see it is much appreciated :-)

Considering this specific issue is extremely low priority it is probably best to wait for the release for the production upgrade.

**Comment by moksharthab** [comment]
Hi Mike,

Greetings of the day!

As per your suggestion and with instructions listed in: https://github.com/Velocidex/velociraptor#getting-the-latest-version...

We have downloaded the pre-RC linux binary (re: pull request #2367) and stood up a dev velociraptor server (0.6.8-4 dev)

Since we have limited resources, and have the production server (0.6.7) running still, we have choosen to run a dev server in parallel on the same physical server but on different listening ports. Here's a quick preview of what we're using for our dev:

Server_URL's / Frontend: 8800
GUI: 8809
API: 8801
Monitoring: 8803
Note: we're using self-signed certs method of deployment

We can see the (dev) server is listening on these alternate ports (via "ss -ltn" command output), and can connect to the dev server's GUI so it appears to be up and functioning. We also have the API server exposed on all interfaces using bind_address 0.0.0.0 bind_port 8801

When we scan both the prod version (0.6.7 port 8001) and dev version (0.6.8 port 8801) using the Nmap script "ssl-enum-ciphers" in both outputs it is listing TLS 1.0, and TLS 1.1 but we expected minimum allowed to be TLS 1.3 when scanning the dev/0.6.8 server instance based on the fix.

Please can you assist on what we may be doing wrong, or what we might check, to proceed with testing it?

**Comment by scudette** [comment]
TLS servers work by advertising a bunch of ciphers and versions and then negotiating a higher one after connection.

You can verify the minimum TLS version that the server is prepared to connect on using the openssl client command:

```
openssl s_client -connect localhost:8000 -tls1
openssl s_client -connect localhost:8000 -tls1_1
openssl s_client -connect localhost:8000 -tls1_2
openssl s_client -connect localhost:8000 -tls1_3
```

You will see the client will refuse to connect on anything less than TLS 1.2

**Comment by scudette** [comment]
Oh I see - you are referring to the API service TLS connection. Yes this will still accept TLS 1.0 

I will reopen this issue.

**Comment by moksharthab** [comment]
Thanks a lot Mike 😊

**Comment by scudette** [comment]
Fixed by #2385
